### DISCUSSION TITLE
Create a `samc-class` object using the resistance and absorption data

### DISCUSSION THREAD

**Discussion by cliu229** [andrewmarx/samc#18] [Q&A] [answered]
I am very sorry, I have deleted the original question, and placed it in the discussion forum.
In running the pythons sample script you provided in the IALE2020 workshop, this section does not continue.

This is the code that runs in the sample script file:
pymu_samc <- samc::samc(resistance = samc_vars$resistance,
                        absorption = samc_vars$absorption,
                        latlon = T,
                        tr_fun = function(x){1/mean(x)})

The following is the error message generated after running the code:
Error in samc::samc(resistance = samc_vars$resistance, absorption = samc_vars$absorption,  : 
  unused arguments (resistance = samc_vars$resistance, latlon = T, tr_fun = function(x) {
    1/mean(x)
})

**Comment by andrewmarx** [comment, marked as answer]
Thanks for moving it here; it'll help others working with older code that encounter this issue.

Starting in version [1.4.0](https://andrewmarx.github.io/samc/news/index.html#samc-1-4-0-2021-06-27), some changes were made to the `samc()` function to make it easier to maintain and add new features. As of version 2.0.0, the old approach was permanently removed, which means some older code needs to be updated. Fortunately, it should be simple.

The parameters for the transition function now need to be grouped together into a list, and that list is then passed to the `samc()` function. For your code, it would look like this:

```
# Setup the details for our transition function
tr <- list(fun = function(x) 1/mean(x), # Function for calculating transition probabilities
           dir = 8, # Directions of the transitions. Either 4 or 8.
           sym = TRUE) # Is the function symmetric?

pymu_samc <- samc::samc(data = samc_vars$resistance, absorption = samc_vars$absorption, tr_args = tr)
```

A couple things to note:
1. `latlon` is no longer relevant; it is automatically determined from the input raster
2. The number of directions, and whether the function is symmetric is now mandatory; the function doesn't assume a default
3. the `resistance` argument to `samc()` no longer exists; it's just `data` now

For that workshop, another issue you might encounter is that lines with `samc@p` no longer work. Ideally, `@` shouldn't be used to access object properties, but at the time it was the only option. Now, you would replace it with `samc$p_matrix`

**Reply by cliu229** [reply]
Awesome! really appreciate your detailed answer, now the problem has been solved！
